osbrain/helper.py

```diff
@@ -113,69 +113,80 @@ def agent_dies(agent, nsproxy, timeout=1.):
     return True
 
 
-def attribute_match(attribute, length=None, data=None, value=None):
+def attribute_match_all(attribute, length=None, data=None, value=None):
     """
-    Check if an attribute matches any condition:
+    Check if an attribute matches all of the following specified conditions:
 
     - Minimum length.
     - Contains an item.
     - Is exactly a value.
 
+    Note that only those conditions explicitly passed will be checked.
+
     Parameters
     ----------
     attribute : anything
         The attribute to match against.
     length : int, default is None
-        Return True if the attribute has this minimum length.
+        If specified, the attribute must reach this length (or higher).
     data : anything, default is None
-        Return True if the attribute contains this value.
+        If specified, the attribute must contain this value.
     value : anything, default is None
-        Return True if the attribute contains this value.
+        If specified, the attribute must be this value.
 
     Returns
     -------
     bool
         Whether the attribute matches any condition.
     """
     assert length is not None or data is not None or value is not None, \
```

Comment:
Note that before it would return False if no conditions where given. That was wrong, hence the assertion. In this case, if no conditions are given, it will return True which, in my opinion, is right. So maybe we don't need the assertion anymore?